test/engine-luatest/pagination_test.lua

```diff
@@ -557,6 +557,19 @@ tree_g.test_empty_page_with_offset = function(cg)
     end)
```

Comment:
Please, enable the cache for this test suite.

Reply:
Well, the cache's enabled by default, but you're right - better enable it explicitly. Done it like this:

<details><summary>diff</summary>

```diff
diff --git a/test/engine-luatest/pagination_test.lua b/test/engine-luatest/pagination_test.lua
index 17eaac125d4a..c295a1df0579 100644
--- a/test/engine-luatest/pagination_test.lua
+++ b/test/engine-luatest/pagination_test.lua
@@ -557,6 +557,21 @@ tree_g.test_empty_page_with_offset = function(cg)
     end)
 end
 
+tree_g.before_test('test_gh_7943', function(cg)
+    -- Vinyl tuple cache must be enabled to reproduce the issue.
+    cg.server:exec(function()
+        rawset(_G, 'vinyl_cache', box.cfg.vinyl_cache)
+        box.cfg({vinyl_cache = 1024 * 1024})
+    end)
+end)
+
+tree_g.after_test('test_gh_7943', function(cg)
+    cg.server:exec(function()
+        box.cfg({vinyl_cache = rawget(_G, 'vinyl_cache')})
+        rawset(_G, 'vinyl_cache', nil)
+    end)
+end)
+
 tree_g.test_gh_7943 = function(cg)
     cg.server:exec(function()
         local t = require('luatest')
```
</details>